File: 350 Intersection of Two Arrays II/350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        map = {}
        res = []
        for x in nums1:
            map[x] = map.get(x,0) + 1
            #map[x] = (map[x] + 1) or 1
        
        for x in nums2:
            value = map.get(x,False)
            if value:
                res.append(x)
                map[x] = value - 1
        return res

File: 350 Intersection of Two Arrays II/350.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    const map = new Map();
    const res = [];
    nums1.forEach(x=> map.set(x, (map.get(x) || 0) + 1));
    
    nums2.forEach(x=> {
        let value = map.get(x);
        if(value) {
            res.push(x);
            map.set(x, value - 1);
        }
    });
    return res;
};
